### PR TITLE
Add scripts to easily destroy tfstate backend

### PR DESCRIPTION
**Pull request by osterman**
## what
* Reorganize scripts into `scripts/` folder
* Standardize around `Makefile` as the task runner
* Add `scripts/destroy.sh` to easily destroy tfstate bucket
* Annotate all scripts to explain what's going on

## why
* Easy destruction of state bucket is needed for CI/CD testing environments

## demo
```
Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
null_resource.default: Refreshing state... (ID: 892065051778034330)
null_resource.default: Refreshing state... (ID: 5821758904588873110)
aws_s3_bucket.default: Refreshing state... (ID: eg-staging-test-state)
aws_dynamodb_table.with_server_side_encryption: Refreshing state... (ID: eg-staging-test-state-lock)
module.tfstate_backend.aws_s3_bucket.default: Destroying... (ID: eg-staging-test-state)
module.tfstate_backend.aws_dynamodb_table.with_server_side_encryption: Destroying... (ID: eg-staging-test-state-lock)
module.tfstate_backend.aws_s3_bucket.default: Destruction complete after 1s
module.tfstate_backend.module.s3_bucket_label.null_resource.default: Destroying... (ID: 892065051778034330)
module.tfstate_backend.module.s3_bucket_label.null_resource.default: Destruction complete after 0s
module.tfstate_backend.aws_dynamodb_table.with_server_side_encryption: Destruction complete after 1s
module.tfstate_backend.module.dynamodb_table_label.null_resource.default: Destroying... (ID: 5821758904588873110)
module.tfstate_backend.module.dynamodb_table_label.null_resource.default: Destruction complete after 0s

Destroy complete! Resources: 4 destroyed.
```

### CODE REVIEW

File: aws/tfstate-backend/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Apply the configuration
+apply:
```
Comment (goruha) [inline aws/tfstate-backend/Makefile]: Rename `apply` to `init`.
`Apply` - confuse that this target can be used for update, but this target should be ran only once on `init`.

Comment (osterman) [reply aws/tfstate-backend/Makefile]: Ok, that's fair.

File: aws/tfstate-backend/scripts/destroy.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+rm -rf .terraform terraform.tfstate
+
+# Init terraform with S3 state enabled. Assumes state was previously initialized.
+init-terraform
+
+# Unmount remote bucket (if mounted)
+s3 unmount
+
+# Store the current state so we can destroy resources without catch-22
+terraform state pull > terraform.tfstate
```
Comment (aknysh) [inline aws/tfstate-backend/scripts/destroy.sh]: do we first unmount the bucket with `s3 unmount` above and then try to pull the remote state from the unmounted bucket?

Comment (osterman) [reply aws/tfstate-backend/scripts/destroy.sh]: Mounting the S3 bucket has no bearing on how state is managed. The bucket is only mounted as a "nice to have" for occasionally writing things like SSH keys somewhere.

**Comment by osterman** [comment]
Updated the `README.md` - please review

**Pull request merged**